device/cuda/include/traccc/cuda/seeding/seed_finding.hpp

```diff
@@ -24,112 +24,129 @@
 
 // System include(s).
 #include <algorithm>
-#include <iostream>
-#include <mutex>
 
 namespace traccc {
 namespace cuda {
 
 /// Seed finding for cuda
 struct seed_finding : public algorithm<host_seed_collection(
-                          host_spacepoint_container&&, sp_grid&&)> {
+                          host_spacepoint_container&&, sp_grid_buffer&&)> {
 
     /// Constructor for the cuda seed finding
     ///
     /// @param config is seed finder configuration parameters
     /// @param sp_grid spacepoint grid
     /// @param mr vecmem memory resource
-    seed_finding(seedfinder_config& config, unsigned int nbins,
-                 vecmem::memory_resource& mr)
-        : m_seedfinder_config(config),
-          m_mr(mr),
-          doublet_counter_container(nbins, &m_mr.get()),
-          mid_bot_container(nbins, &m_mr.get()),
-          mid_top_container(nbins, &m_mr.get()),
-          triplet_counter_container(nbins, &m_mr.get()),
-          triplet_container(nbins, &m_mr.get()),
-          seed_collection(&m_mr.get()) {}
+    seed_finding(seedfinder_config& config, vecmem::memory_resource& mr)
```

Comment:
It should really receive a const reference to the configuration object. Requiring a non-const reference is just asking for trouble.

Reply:
Right